vmdb/app/helpers/application_helper.rb

```diff
@@ -172,6 +174,11 @@ def view_to_url(view, parent=nil)
           elsif ["Vm"].include?(view.db) && parent && request.parameters[:controller] != "vm"
             # this is to handle link to a vm in vm explorer from service explorer
             return url_for(:controller=>"vm_or_template", :action=>"show") + "/"
+          elsif %w(ConfigurationProfileForeman).include?(view.db) &&
+                request.parameters[:controller] == "provider_foreman"
+            return url_for(:action => action, :id => nil) + "/"
+          elsif %w(ConfiguredSystemForeman).include?(view.db) && request.parameters[:controller] == "provider_foreman"
+            return url_for(:action => action, :id => nil) + "/"
```

Comment:
Can we at least pull out the `if @explorer` branch into a `view_to_url` method on a SomeExplorer class or something/anything?  Let's try to make this method slightly less complicated not more.  Anything...